src/ripple/app/ledger/impl/BuildLedger.cpp

```diff
@@ -0,0 +1,240 @@
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/ledger/BuildLedger.h>
+#include <ripple/app/ledger/Ledger.h>
+#include <ripple/app/ledger/LedgerReplay.h>
+#include <ripple/app/ledger/OpenLedger.h>
+#include <ripple/app/main/Application.h>
+#include <ripple/app/misc/CanonicalTXSet.h>
+#include <ripple/app/tx/apply.h>
+#include <ripple/protocol/Feature.h>
+
+namespace ripple {
+
+/* Generic buildLedgerImpl that dispatches to ApplyTxs invocable with signature
+    void(OpenView&, std::shared_ptr<Ledger> const&)
+   It is responsible for adding transactions to the open view to generate the
+   new ledger. It is generic since the mechanics differ for consensus
+   generated ledgers versus replayed ledgers.
+*/
+template <class ApplyTxs>
+std::shared_ptr<Ledger>
+buildLedgerImpl(
+    std::shared_ptr<Ledger const> const& parent,
+    NetClock::time_point closeTime,
+    const bool closeTimeCorrect,
+    NetClock::duration closeResolution,
+    Application& app,
+    beast::Journal j,
+    ApplyTxs&& applyTxs)
+{
+    auto buildLCL = std::make_shared<Ledger>(*parent, closeTime);
+
+    auto const v2_enabled = buildLCL->rules().enabled(featureSHAMapV2);
+    auto v2_transition = false;
+    if (v2_enabled && !buildLCL->stateMap().is_v2())
+    {
+        buildLCL->make_v2();
+        v2_transition = true;
+    }
+
+    // Set up to write SHAMap changes to our database,
+    //   perform updates, extract changes
+
+    {
+        OpenView accum(&*buildLCL);
+        assert(!accum.open());
+        applyTxs(accum, buildLCL);
+        accum.apply(*buildLCL);
+    }
+
+    buildLCL->updateSkipList();
+
+    {
+        // Write the final version of all modified SHAMap
+        // nodes to the node store to preserve the new LCL
+
+        int asf = buildLCL->stateMap().flushDirty(
+            hotACCOUNT_NODE, buildLCL->info().seq);
+        int tmf = buildLCL->txMap().flushDirty(
+            hotTRANSACTION_NODE, buildLCL->info().seq);
+        JLOG(j.debug()) << "Flushed " << asf << " accounts and " << tmf
+                        << " transaction nodes";
+    }
+    buildLCL->unshare();
+
+    // Accept ledger
+    buildLCL->setAccepted(
+        closeTime, closeResolution, closeTimeCorrect, app.config());
+
+    return buildLCL;
+}
+
+/** Apply a set of consensus transactions to a ledger.
+
+  Typically the txFilter is used to reject transactions
```

Comment:
There no longer is a `txFilter` argument passed to this function, so this comment is out of date.